Documentation fixes in serialization.py

PradyumnaKrishna opened the pull request:
Issue: https://github.com/in-toto/in-toto/pull/503#pullrequestreview-1099354277

### Description of the changes being introduced by the pull request:

Removing references to `Serializable` from doc string of SerializationMixin and mention of default serializer/deserializer from docs.

Changing method name of default_serializer to _default_serializer i.e. protected element of class. Also, avoid getting generated in sphinx docs.

### Please verify and check that the pull request fulfils the following requirements:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

PradyumnaKrishna committed:
Documentation fixes in serialization.py

Removing references to `Serializable` from doc string of
SerializationMixin and mention of default serializer/deserializer
from docs.

Changing method name of default_serializer to _default_serializer
i.e. protected element of class. Also, avoid getting generated in
sphinx docs.

Signed-off-by: Pradyumna Krishna <git@onpy.in>

lukpueh reviewed (changes requested):
From https://github.com/in-toto/in-toto/pull/503#pullrequestreview-1099354277:

> - please use [...] full import paths for class names etc., eg. ``securesystemslib.serialization.JSONDeserializer``

Generally, I think it's okay to not use the full import path in a docstring, but only if it's clear from the context, where the name comes from. Given that some of these docstrings will be shown on in-toto/python-tuf readthedocs, we should give the reader a hint to look for those names in securesystemslib.

PradyumnaKrishna committed:
Add reference for classes mentioned in doc-string

Updated doc-string to provide hints, where look for the mentioned
class. For example ``JSONSerializer`` is not renamed to
``securesystemslib.serialization.JSONSerializer``.

Signed-off-by: Pradyumna Krishna <git@onpy.in>
